core/src/env.rs

```diff
@@ -192,12 +192,12 @@ pub trait Externalities {
         mem: &mut impl Memory,
     ) -> Result<WasmPage, Self::AllocError>;
 
-    /// Free specific memory page.
-    ///
-    /// Unlike traditional allocator, if multiple pages allocated via `alloc`, all pages
-    /// should be `free`-d separately.
+    /// Free specific memory page
     fn free(&mut self, page: WasmPage) -> Result<(), Self::AllocError>;
```

Comment:
You can declare default realisation using Self::free_range just right here

Reply:
done, thanks